src/util/address.rs

```diff
@@ -511,13 +513,34 @@ impl Address {
         }
     }
 
-    /// Create a pay to taproot address
-    pub fn p2tr(taptweaked_key: schnorrsig::PublicKey, network: Network) -> Address {
+    /// Create a pay to taproot address from untweaked key
+    pub fn p2tr<C: Verification>(
+        secp: Secp256k1<C>, 
+        internal_key: UntweakedPublicKey, 
+        merkle_root: Option<TapBranchHash>, 
+        network: Network
+    ) -> Address {
         Address {
             network: network,
             payload: Payload::WitnessProgram {
                 version: WitnessVersion::V1,
-                program: taptweaked_key.serialize().to_vec()
+                program: internal_key.tap_tweak(secp, merkle_root).into_inner().serialize().to_vec()
+            }
+        }
+    }
+
+    /// Create a pay to taproot address from a tweaked key.
+    /// 
+    /// This method is dangerous and can lead to loss of funds if used incorrectly.
```

Comment:
The method is no longer dangerous.

Reply:
I would like to highlight that this method is not recommended to use. Users should use p2tr() whenever possible.

In the future, we expect backups to backup only back the internal key, so I would prefer that whenever users obtain the other API to avoid complications there. We don't want to confuse the internal secret key without the output key secret key.

Reply:
If `TweakedPublicKey` doesn't provide method for serializing it should be impossible to accidentally use it in backups?

Anyway, I agree with the current wording saying it's not recommended but without `dangerous` part.